Exceptions everytime I try to work with a document that contains a binary

I've got some documents that contains binaries, when I try to work with them I get this error:

```
panic: runtime error: comparing uncomparable type primitive.Binary [recovered]
        panic: runtime error: comparing uncomparable type primitive.Binary

goroutine 1 [running]:
github.com/kopecmaciej/tview.(*Application).Run.func1()
        github.com/kopecmaciej/tview@v0.1.4/application.go:297 +0x45
panic({0xc70da0?, 0xc0003018a0?})
        runtime/panic.go:792 +0x132
github.com/kopecmaciej/vi-mongo/internal/mongo.(*CollectionState).GetDocById(0x0?, {0xcd3ba0, 0xc000bf0340})
        github.com/kopecmaciej/vi-mongo/internal/mongo/state.go:37 +0x15a
github.com/kopecmaciej/vi-mongo/internal/mongo.(*CollectionState).GetJsonDocById(0x497fe5?, {0xcd3ba0?, 0xc000bf0340?})
        github.com/kopecmaciej/vi-mongo/internal/mongo/state.go:46 +0x27
github.com/kopecmaciej/vi-mongo/internal/tui/component.(*Peeker).Render(0xc00032c3f0, {0xe8f850, 0x14f6420}, 0xc00029fb00, {0xcd3ba0, 0xc000bf0340})
        github.com/kopecmaciej/vi-mongo/internal/tui/component/peeker.go:137 +0x56
github.com/kopecmaciej/vi-mongo/internal/tui/component.(*Content).handlePeekDocument(0xc00029e880, {0xe8f850, 0x14f6420}, 0xc000300830?, 0x1?)
        github.com/kopecmaciej/vi-mongo/internal/tui/component/content.go:724 +0x65
github.com/kopecmaciej/vi-mongo/internal/tui/component.(*Content).setKeybindings.func1(0xc0001f4820)
        github.com/kopecmaciej/vi-mongo/internal/tui/component/content.go:219 +0x1725
github.com/kopecmaciej/vi-mongo/internal/tui/core.(*Table).InputHandler.(*Table).InputHandler.(*Box).WrapInputHandler.func2(0x419885?, 0xc000301840)
        github.com/kopecmaciej/tview@v0.1.4/box.go:181 +0x37
github.com/kopecmaciej/vi-mongo/internal/tui/core.(*Flex).InputHandler.(*Flex).InputHandler.func1(0xc0001f4820, 0xc000301840)
        github.com/kopecmaciej/tview@v0.1.4/flex.go:265 +0xd7
github.com/kopecmaciej/vi-mongo/internal/tui/core.(*Flex).InputHandler.(*Flex).InputHandler.(*Box).WrapInputHandler.func2(0xc00032c2a0?, 0x18?)
        github.com/kopecmaciej/tview@v0.1.4/box.go:184 +0x50
github.com/kopecmaciej/vi-mongo/internal/tui/component.(*Content).InputHandler.(*Flex).InputHandler.func1(0xc0001f4820, 0xc000301840)
        github.com/kopecmaciej/tview@v0.1.4/flex.go:265 +0xd7
github.com/kopecmaciej/vi-mongo/internal/tui/component.(*Content).InputHandler.(*Flex).InputHandler.(*Box).WrapInputHandler.func2(0xc00032c270?, 0x18?)
        github.com/kopecmaciej/tview@v0.1.4/box.go:184 +0x50
github.com/kopecmaciej/vi-mongo/internal/tui/core.(*Flex).InputHandler.(*Flex).InputHandler.func1(0xc0001f4820, 0xc000301840)
        github.com/kopecmaciej/tview@v0.1.4/flex.go:265 +0xd7
github.com/kopecmaciej/vi-mongo/internal/tui/core.(*Flex).InputHandler.(*Flex).InputHandler.(*Box).WrapInputHandler.func2(0xc0002ebf20?, 0x0?)
        github.com/kopecmaciej/tview@v0.1.4/box.go:184 +0x50
github.com/kopecmaciej/vi-mongo/internal/tui/page.(*Main).InputHandler.(*Flex).InputHandler.func1(0xc0001f4820, 0xc000301840)
        github.com/kopecmaciej/tview@v0.1.4/flex.go:265 +0xd7
github.com/kopecmaciej/vi-mongo/internal/tui/page.(*Main).InputHandler.(*Flex).InputHandler.(*Box).WrapInputHandler.func2(0xc0002ebef0?, 0xc000301840?)
        github.com/kopecmaciej/tview@v0.1.4/box.go:184 +0x50
github.com/kopecmaciej/vi-mongo/internal/tui/core.(*Pages).InputHandler.(*Pages).InputHandler.func1(0xc0001f4820, 0xc000301840)
        github.com/kopecmaciej/tview@v0.1.4/pages.go:323 +0xa2
github.com/kopecmaciej/vi-mongo/internal/tui/core.(*Pages).InputHandler.(*Pages).InputHandler.(*Box).WrapInputHandler.func2(0xc0001f4820?, 0xc00068fa60?)
        github.com/kopecmaciej/tview@v0.1.4/box.go:184 +0x50
github.com/kopecmaciej/tview.(*Application).Run(0xc000328000)
        github.com/kopecmaciej/tview@v0.1.4/application.go:416 +0x938
github.com/kopecmaciej/vi-mongo/internal/tui.(*App).Run(...)
        github.com/kopecmaciej/vi-mongo/internal/tui/app.go:59
github.com/kopecmaciej/vi-mongo/cmd.runApp(0x14ccb40, {0xd5cae0?, 0x4?, 0xd5cae4?})
        github.com/kopecmaciej/vi-mongo/cmd/cmd.go:171 +0x3c5
github.com/spf13/cobra.(*Command).execute(0x14ccb40, {0xc00012a140, 0x0, 0x0})
        github.com/spf13/cobra@v1.8.1/command.go:989 +0xabb
github.com/spf13/cobra.(*Command).ExecuteC(0x14ccb40)
        github.com/spf13/cobra@v1.8.1/command.go:1117 +0x44f
github.com/spf13/cobra.(*Command).Execute(...)
        github.com/spf13/cobra@v1.8.1/command.go:1041
github.com/kopecmaciej/vi-mongo/cmd.Execute(...)
        github.com/kopecmaciej/vi-mongo/cmd/cmd.go:39
main.main()
        github.com/kopecmaciej/vi-mongo/main.go:10 +0x1a
```

**How to reproduce:**
1: create a collection
2: add a document with a binary field.
3: open the collection
4: visit any of the documents with binaries.

--- a/internal/mongo/state.go
+++ b/internal/mongo/state.go
@@ -34,7 +34,7 @@ func (c *CollectionState) GetAllDocs() []primitive.M {
 func (c *CollectionState) GetDocById(id any) primitive.M {
 	for _, doc := range c.docs {
 		if reflect.TypeOf(doc["_id"]) == reflect.TypeOf(id) {
-			if doc["_id"] == id {
+			if reflect.DeepEqual(doc["_id"], id) {
 				return util.DeepCopy(doc)
 			}
 		}
@@ -131,7 +131,7 @@ func (c *CollectionState) UpdateRawDoc(doc string) error {
 		return err
 	}
 	for i, existingDoc := range c.docs {
-		if existingDoc["_id"] == docMap["_id"] {
+		if reflect.DeepEqual(existingDoc["_id"], docMap["_id"]) {
 			c.docs[i] = docMap
 			return nil
 		}
@@ -147,7 +147,7 @@ func (c *CollectionState) AppendDoc(doc primitive.M) {
 
 func (c *CollectionState) DeleteDoc(id any) {
 	for i, doc := range c.docs {
-		if doc["_id"] == id {
+		if reflect.DeepEqual(doc["_id"], id) {
 			c.docs = append(c.docs[:i], c.docs[i+1:]...)
 			c.Count--
 			return

--- a/internal/mongo/state_test.go
+++ b/internal/mongo/state_test.go
@@ -117,3 +117,50 @@ func TestCollectionState_GetJsonDocById_DoesNotModifyState(t *testing.T) {
 	assert.Equal(t, primitive.M{"_id": id1, "value": 1}, cs.docs[0])
 	assert.Equal(t, primitive.M{"_id": id2, "value": 2}, cs.docs[1])
 }
+
+func TestCollectionState_GetDocById_WithBinaryId(t *testing.T) {
+	binaryId := primitive.Binary{Data: []byte{1, 2, 3, 4}, Subtype: 0}
+	cs := &CollectionState{
+		docs: []primitive.M{
+			{"_id": binaryId, "value": "binary_doc"},
+		},
+	}
+
+	doc := cs.GetDocById(binaryId)
+	assert.NotNil(t, doc)
+	assert.Equal(t, binaryId, doc["_id"])
+	assert.Equal(t, "binary_doc", doc["value"])
+
+	differentBinaryId := primitive.Binary{Data: []byte{5, 6, 7, 8}, Subtype: 0}
+	doc = cs.GetDocById(differentBinaryId)
+	assert.Nil(t, doc)
+}
+
+func TestCollectionState_DeleteDoc_WithBinaryId(t *testing.T) {
+	binaryId := primitive.Binary{Data: []byte{1, 2, 3, 4}, Subtype: 0}
+	cs := &CollectionState{
+		docs: []primitive.M{
+			{"_id": binaryId, "value": "binary_doc"},
+		},
+		Count: 1,
+	}
+
+	cs.DeleteDoc(binaryId)
+	assert.Len(t, cs.docs, 0)
+	assert.Equal(t, int64(0), cs.Count)
+}
+
+func TestCollectionState_UpdateRawDoc_WithBinaryId(t *testing.T) {
+	binaryId := primitive.Binary{Data: []byte{1, 2, 3, 4}, Subtype: 0}
+	cs := &CollectionState{
+		docs: []primitive.M{
+			{"_id": binaryId, "value": "old_value"},
+		},
+	}
+
+	updatedDoc := `{"_id": {"$binary": {"base64": "AQIDBA==", "subType": "00"}}, "value": "new_value"}`
+	err := cs.UpdateRawDoc(updatedDoc)
+	assert.NoError(t, err)
+	assert.Len(t, cs.docs, 1)
+	assert.Equal(t, "new_value", cs.docs[0]["value"])
+}

